src/client/common/process/environmentActivationService.ts

```diff
@@ -225,6 +225,12 @@ export class EnvironmentActivationService implements IEnvironmentActivationServi
             // Remove it from cache (if it exists).
             void this.memento.update(key, undefined);
         }
+        if (env && customEnvVars) {
```

Comment:
In debt will refactor this, its a bigger change, but I think the way we create Python execution services is very complicated now and we've regressed in certain areas hence the need to fix this change in a number of places now.